x-pack/legacy/plugins/maps/public/layers/sources/ems_file_source/update_source_editor.js

```diff
@@ -36,16 +39,16 @@ export class UpdateSourceEditor extends Component {
       const emsFiles = await emsClient.getFileLayers();
       const emsFile = emsFiles.find((emsFile => emsFile.getId() === this.props.layerId));
       const emsFields = emsFile.getFieldsInLanguage();
-      fields = emsFields.map(field => {
-        return {
-          name: field.name,
-          label: field.description
-        };
-      });
+      fields = emsFields.map(field => new EMSFileField({
```

Comment:
How about passing EMSFileSource.createField in as a props so the creating of the field is not duplicated?